dotCMS/src/main/java/com/dotcms/rendering/velocity/services/VTLLoader.java

```diff
@@ -0,0 +1,144 @@
+            if(UtilMethods.isSet(Config.getStringProperty("ASSET_REAL_PATH"))){
+                f = new File(Config.getStringProperty("ASSET_REAL_PATH"));
+                if(f.exists()){
+                        assetRealCanoncalPath = f.getCanonicalPath();
+                }
+            }
+        } catch (IOException e) {
+            Logger.fatal(this,e.getMessage(),e);
+        }
+
+        try {
+            if(UtilMethods.isSet(Config.getStringProperty("ASSET_PATH"))){
+                f = new File(FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH")));
+                if(f.exists()){
+                    assetCanoncalPath = f.getCanonicalPath();
+                }
+            }
+        } catch (IOException e) {
+            Logger.fatal(this,e.getMessage(),e);
+        }
+
+
+    }
+
+    InputStream streamFile(String filePath) throws IOException {
+        boolean serveFile = false;
+        Logger.debug(this, "Not a CMS Velocity File : " + filePath);
+
+        java.io.File f = null;
+        String lookingFor = "";
+        if (filePath.startsWith("dynamic")) {
+            lookingFor = ConfigUtils.getDynamicContentPath() + File.separator + "velocity" + File.separator + filePath;
+
+        } else {
+            lookingFor = VELOCITY_ROOT + filePath;
+        }
+        f = new java.io.File(lookingFor);
+        if (!f.exists()) {
+            f = new java.io.File(filePath);
+        }
+        if (!f.exists()) {
+            throw new ResourceNotFoundException("cannot find resource");
+        }
+        String canon = f.getCanonicalPath();
+        File dynamicContent = new File(ConfigUtils.getDynamicContentPath());
+
+        if (assetRealCanoncalPath != null && canon.startsWith(assetRealCanoncalPath)) {
+            serveFile = true;
+        } else if (velocityCanoncalPath != null && canon.startsWith(velocityCanoncalPath)) {
+            serveFile = true;
+        } else if (assetCanoncalPath != null && canon.startsWith(assetCanoncalPath)) {
+            serveFile = true;
+        } else if (canon.startsWith(dynamicContent.getCanonicalPath())) {
+            serveFile = true;
+        }
+        if (!serveFile) {
+            Logger.warn(this, "POSSIBLE HACK ATTACK DotResourceLoader: " + lookingFor);
+            throw new ResourceNotFoundException("cannot find resource");
+        }
+        return new BufferedInputStream(Files.newInputStream(f.toPath()));
+
+    }
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, boolean live, String language, String filePath) {
+        try {
+            return streamFile(filePath);
+        } catch (IOException e) {
+            throw new ResourceNotFoundException("cannot find resource:" + filePath);
+        }
+    }
+
+    @Override
+    public void invalidate(Object obj) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void invalidate(Object obj, boolean live) {
+        // TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)